sdk/helper/certutil/types.go

```diff
@@ -368,14 +368,37 @@ func (p *ParsedCertBundle) Verify() error {
 	certPath := p.GetCertificatePath()
 	if len(certPath) > 1 {
 		for i, caCert := range certPath[1:] {
-			if !caCert.Certificate.IsCA {
+			child := certPath[i].Certificate
+			parent := caCert.Certificate
+			if !parent.IsCA {
 				return fmt.Errorf("certificate %d of certificate chain is not a certificate authority", i+1)
 			}
-			if !bytes.Equal(certPath[i].Certificate.AuthorityKeyId, caCert.Certificate.SubjectKeyId) {
+
+			// Some CAs do not provision AKID/SKID, so we must resort to
+			// checking the issuer/subject link and validating the signature.
+			//
+			// However, when AKID is provisioned, we trust and validate it.
+			// This is a bit more involved, because some CAs like Microsoft AD
+			// have apparently changed techniques for encoding SKID/AKID over
```

Comment:
Revise this statement, they seem to be internally consistent upon further inspection.